pyracf/user/user_admin.py

```diff
@@ -765,13 +768,33 @@ def set_tso_data_set_allocation_unit(
     # ============================================================================
     def add(self, userid: str, traits: dict = {}) -> Union[dict, bytes]:
         """Create a new user."""
-        self._build_segment_dictionaries(traits)
-        user_request = UserRequest(userid, "set")
-        self._build_xml_segments(user_request)
-        return self._make_request(user_request)
-
-    def alter(self, userid: str, traits: dict = {}) -> Union[dict, bytes]:
+        if self._generate_requests_only:
+            self._build_segment_dictionaries(traits)
+            user_request = UserRequest(userid, "set")
+            self._build_xml_segments(user_request)
+            return self._make_request(user_request)
+        try:
+            self.extract(userid)
+        except SecurityRequestError as exception:
+            if not exception.contains_error_message(self._profile_type, "ICH30001I"):
+                raise exception
+            self._build_segment_dictionaries(traits)
+            user_request = UserRequest(userid, "set")
+            self._build_xml_segments(user_request)
+            return self._make_request(user_request)
+        raise AddOperationError(userid, self._profile_type)
+
+    def alter(self, userid: str, traits: dict) -> Union[dict, bytes]:
```

Comment:
I am guessing that `traits: dict = {}` was changed to `traits: dict` for alter because giving no traits now causes an error now? Even through having `traits` as a keyword argument that is technically option can result in nothing happening, I set it up that way to make the syntax consistent.

Reply:
It doesn't cause an error, but it serves no purpose. Previously it could potentially be used to add a profile, but now an empty alter accomplishes nothing and is more expensive than an extract. I can add the option back in to keep syntax the same, but realistically, I think that it would only ever be done by accident, so having pyracf yell over a non-optional argument seems like the best case to me.

Reply:
Sure, I think it will still be good to document the `traits=` syntax, but we can make it so that it fails if the user does not provide the `traits` argument.